app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,52 @@
+class PostsController < ApplicationController
+
+  before_action :fetch_post, only: [:edit, :update, :destroy]
+
+  def index
+    @posts = Post.all.limit(5)
+    @post = @posts.max_by{|p| p.created_at}
+    @posts = @posts - [@post]
+  end
+
+  def show
+    @posts = Post.limit(4).where("id != ?", params[:id].to_i)
+    @post = Post.find(params[:id].to_i)
```

Comment:
With the ActiveRecord `find`, you needn't change the `params[:id]` into an integer.